tests/models/sam_hq/test_modeling_sam_hq.py

```diff
@@ -0,0 +1,863 @@
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores
+
+        masks = outputs.pred_masks[0, 0, 0, 0, :3]
+        self.assertTrue(torch.allclose(scores[0][0][-1], torch.tensor(0.4482), atol=2e-4))
+        self.assertTrue(
+            torch.allclose(masks, torch.tensor([-13.1695, -14.6201, -14.8989]).to(torch_device), atol=2e-4)
+        )
+
+    def test_inference_mask_generation_one_point_one_bb(self):
+        model = SamHQModel.from_pretrained("sushmanth/sam_hq_vit_b")
+        processor = SamHQProcessor.from_pretrained("sushmanth/sam_hq_vit_b")
+
+        model.to(torch_device)
+        model.eval()
+
+        raw_image = prepare_image()
+        input_boxes = [[[650, 900, 1000, 1250]]]
+        input_points = [[[820, 1080]]]
+
+        inputs = processor(
+            images=raw_image, input_boxes=input_boxes, input_points=input_points, return_tensors="pt"
+        ).to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze()
+        masks = outputs.pred_masks[0, 0, 0, 0, :3]
+        self.assertTrue(torch.allclose(scores[-1], torch.tensor(0.9700), atol=2e-4))
+        self.assertTrue(
+            torch.allclose(masks, torch.tensor([-29.9144, -30.0546, -30.9526]).to(torch_device), atol=2e-4)
+        )
+
+    def test_inference_mask_generation_batched_points_batched_images(self):
+        model = SamHQModel.from_pretrained("sushmanth/sam_hq_vit_b")
+        processor = SamHQProcessor.from_pretrained("sushmanth/sam_hq_vit_b")
+
+        model.to(torch_device)
+        model.eval()
+
+        raw_image = prepare_image()
+        input_points = [
+            [[[820, 1080]], [[820, 1080]], [[820, 1080]], [[820, 1080]]],
+            [[[510, 1080]], [[820, 1080]], [[820, 1080]], [[820, 1080]]],
+        ]
+
+        inputs = processor(images=[raw_image, raw_image], input_points=input_points, return_tensors="pt").to(
+            torch_device
+        )
+
+        with torch.no_grad():
+            outputs = model(**inputs)
+        scores = outputs.iou_scores.squeeze().cpu()
+        masks = outputs.pred_masks[0, 0, 0, 0, :3].cpu()
+        EXPECTED_SCORES = torch.tensor(
+            [
+                [
+                    [0.9195, 0.8316, 0.6614],
+                    [0.9195, 0.8316, 0.6614],
+                    [0.9195, 0.8316, 0.6614],
+                    [0.9195, 0.8316, 0.6614],
+                ],
+                [
+                    [0.7598, 0.7388, 0.3110],
+                    [0.9195, 0.8317, 0.6614],
+                    [0.9195, 0.8317, 0.6614],
+                    [0.9195, 0.8317, 0.6614],
+                ],
+            ]
+        )
+        EXPECTED_MASKS = torch.tensor([-40.2445, -37.4300, -38.1577])
+
+        self.assertTrue(torch.allclose(scores, EXPECTED_SCORES, atol=1e-3))
+        self.assertTrue(torch.allclose(masks, EXPECTED_MASKS, atol=1e-3))
```

Comment:
For instance this seems needed depending on the runner
```suggestion
        self.assertTrue(torch.allclose(masks, EXPECTED_MASKS, atol=9e-3))
```
which surprises me a bit. Again, the results look fine and are in the range of 1e2 so `atol` is not the most relevant, `rtol` would be better since masks are not normalized values.